actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -800,7 +806,7 @@ def permitted=(new_permitted)
       end
 
       def fields_for_style?
-        @parameters.all? { |k, v| k =~ /\A-?\d+\z/ && (v.is_a?(Hash) || v.is_a?(Parameters)) }
+        @parameters.any? { |k, v| Parameters.field_style_valid?(k, v) }
```

Comment:
This makes sense, we're only checking if one or more a numeric. Maybe we should rename this method (and the class level one too) to `#nested_attributes?` or `#nested?` instead. I think `#field_for_style?` is too married to the concept of form data.